src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_DeconstructionAssignmentOperator.cs

```diff
@@ -368,9 +391,19 @@ internal BoundExpression PopLast()
             }
 
             // This can only be called after Consolidate
-            internal ImmutableArray<BoundExpression> ToImmutableAndFree()
+            internal ImmutableArray<BoundExpression> ToImmutableAndClear()
             {
```

Comment:
>{ [](start = 12, length = 1)

Can we assert that `deconstructions`, `conversions` and `assignments` are cleared/empty?